x/ecocredit/server/server.go

```diff
@@ -157,6 +173,19 @@ func newServer(storeKey sdk.StoreKey, paramSpace paramtypes.Subspace,
 	if err != nil {
 		panic(err.Error())
 	}
+	s.buyOrderByExpirationIndex, err = orm.NewIndex(buyOrderTableBuilder, BuyOrderByExpirationIndexPrefix, func(value interface{}) ([]interface{}, error) {
+		order, ok := value.(*ecocredit.BuyOrder)
+		if !ok {
+			return nil, sdkerrors.ErrInvalidType.Wrapf("expected %T got %T", ecocredit.BuyOrder{}, value)
+		}
+		if order.Expiration == nil {
+			return nil, nil
+		}
+		return []interface{}{uint64(order.Expiration.UnixNano())}, nil
+	}, uint64(0))
+	if err != nil {
+		panic(err.Error())
+	}
```

Comment:
Other way to avoid the round trip with de-serialization of index, would be to add a delete method to the index (we can make a task for that). CC: @aaronc 

context: https://github.com/regen-network/regen-ledger/pull/671#discussion_r785142462